learn/guides/ballerina-openapi-support.md

```diff
@@ -0,0 +1,314 @@
+```bash
+$ bal openapi -i <openapi contract> [-n |--nullable]
+```
+
+**See more command options :**
+[OpenAPI to Ballerina Command Reference](#openapi-to-ballerina-command-reference)
+
+
+## Contributing to WSO2 `openapi-connectors` Repository
+If you are willing to see your new client in ballerina central in the future, please follow the below steps and send a GitHub Pull Request.
+
+#### **Step 01 :** Set up the prerequisites 
+1. Fork and clone the [openapi-connectors](https://github.com/ballerina-platform/openapi-connectors) repository. Connectors generated from the Ballerina OpenAPI tool are managed at this repository. 
+
+#### **Step 02 :** Add Ballerina client
+1. Open the clone repository and navigate to `/openapi` directory.
+2. Run `bal new <connector_name> -t lib` to create a new Ballerina package. 
+3. Copy the [generated files](#generate-client-from-openapi-specification)(**client.bal, types.bal, utils.bal**) into the <connector_name> directory. 
+4. Run `bal build` to check whether the newly generated connector is compiling successfully. 
+5. Run necessary unit tests to verify the functionality of the generated connector. 
+6. Add license header at the top of each `.bal` file. 
+7. Add `Package.md` ([guide here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)) and `Module.md` ([guide here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78))
+8. Add connector icon to the root of the connector. Icon needs to be a `.png` of 200x200 px size with name `icon.png`. 
+9. Update `Ballerina.toml` file referring to the [guide here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72).
+
+#### **Step 03:** Send PR to openapi-connectors
+1. Run `./gradlew build`. 
+2. Push the changes and create a Pull Request to the master branch of the `openapi-connectors` repository.
+
+## OpenAPI to Ballerina Command Reference
+```bash
+bal openapi [-i | --input] <openapi-contract-file-path> 
+            [-o | --output] <output-location>
+            [--mode <mode-type>]
+            [--tags <tag-names>] 
+            [--operations <operation-names>] 
+            [-n | --nullable]
+            [--license] <license-file-path> 
+            [--with-tests]
+```
+
+- ##### -i | - -input
+Parameter **openapi-contract-path** specifies the path of the OpenAPI contract file (e.g., `my-api.yaml` or `my-api.json`) and is mandatory.
+- ##### -o | - -output
+The Ballerina files will be generated at the same location OpenAPI command is executed. Optionally, you can point to another directory location using the optional flag `(-o|--output)`.
+- ##### - -mode
+Mode type is optional and can be service or client. The Ballerina service and client will be generated according to the mode. Without `--mode` it will generate both service and client stub for the given openAPI contract.
```

Comment:
```suggestion
Mode type is optional and can be either a service or client. The Ballerina service and client will be generated according to the mode. Without the `--mode`, it will generate both service and client stubs for the given OpenAPI contract.
```